docs/BEST_PRACTICES.md

```diff
@@ -2,11 +2,15 @@
 This is a collection of tips, advice, gotchas and other best practices for using the [Jenkins Pipeline plugin](https://github.com/jenkinsci/workflow-plugin/blob/master/README.md). Contributions and comments are happily accepted.
 
 # General tips and advice
-* Do everything that makes sense there within `stage`s. This will make your builds easier to visualize, debug, etc.
+* Do everything that makes sense there within `stage`s. This will make your builds easier to visualize, debug (e.g it isolates failures to a specific stage), display data (e.g it gets the execution time per build and per stage), etc.
 * Do all real work that involves running a shell script, building, etc, within `node` blocks, so that it actually happens on a real executor, rather than a flyweight executor on the master node.
-* Get your flows from source control - `Jenkinsfile`s, loading libraries, global [CPS](https://en.wikipedia.org/wiki/Continuation-passing_style) library, you name it - but if you pull the main flow from SCM (i.e., Multibranch with `Jenkinsfile`s or `Pipeline from SCM`), be aware that you may need to whitelist a lot of method calls in the script security plugin. [JENKINS-28178](https://issues.jenkins-ci.org/browse/JENKINS-28178) has been created to simplify this, but you'll still want to be careful with that functionality due to the open security it implies.
-* `input` shouldn’t be done within a `node` block - that eats up two executors waiting on the `input` feedback, both the flyweight executor that’s used for the `input` itself and the real executor used by the `node` block, which won’t free up until after the `input` itself has completed.
-
+* Get your flows from source control - `Jenkinsfile`s, loading libraries, global [CPS](https://en.wikipedia.org/wiki/Continuation-passing_style) library, you name it - but if you pull the main flow from SCM (i.e., Multibranch with `Jenkinsfile`s or `Pipeline from SCM`), be aware that you may need to whitelist a lot of method calls in the script security plugin. [JENKINS-28178](https://issues.jenkins-ci.org/browse/JENKINS-28178) has been created to simplify this, but you'll still want to be careful with that functionality due to the open security it implies. By getting your flows from source control, you benefit from `Jenkinsfile` versioning and also testing and merging against your CD Pipeline definition.
```

Comment:
I would not even mention JENKINS-28178 - my hope is that it never comes to pass, because that is a horribly insecure model since it implies anybody with access to the code repo you're building can root your master. 

The *right* way to do this is with a global shared library, that lives in a separately-secured, blessed repo.  At least this way you have isolation between people who have admin level access to the master vs. general developers who can push code.

Reply:
Good point about versioning though.

Reply:
done